Mapping Canonical Cloud Resources ID to OpenTelemetry resource attributes (#33487)
DataDog/documentation: content/en/opentelemetry/mapping/host_metadata.md

**Note:** You need to add these processors and receivers in the Collector running on the host that you want to monitor. A gateway host does not collect this information from remote hosts.

## Canonical cloud resource IDs

Canonical cloud resource IDs (CCRIDs) are cloud provider-assigned resource IDs that uniquely identify a cloud resource. CCRIDs, once added across different observability types, can be used to consistently link different types of data for a given cloud resource. CCRIDs can be added in the same format across all cloud resource types. Widespread addition and adoption of CCRIDs gives access to a variety of use cases across customers and internal teams.

> Enable to seamlessly jump between resources and their associated metrics, traces, and logs for all resource types, eliminating context switching and giving users an end-to-end view of their resources within the same workflow.

Reviewer: ```suggestion
Enable CCRIDs to jump between resources and their associated metrics, traces, and logs for all resource types, eliminating context switching and giving you an end-to-end view of your resources within the same workflow.
```

I'm assuming the "user" here is our customer, rather than the customer's users. If that's incorrect, you can skip this suggestion.

See below for the list of identifier formats per-cloud:
| Cloud   | Identifier Type    | Example                                                                                                                                      |
|---------|--------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
| AWS     | ARN                | `arn:aws:sns:us-east-1:123456789012:instance/example-sns-topic-name`                                                                                  |
| Azure   | Resource ID        | `/subscriptions/0b62a232-b8db-4380-9da6-640f7272ed6d/resourcegroups/lfotriggertest/providers/microsoft.web/sites/resources-task-19cb7afdcbbc`|
| GCP     | CAI Resource Name  | `//file.googleapis.com/projects/datadog-sandbox/locations/us-central1/backups/kevin-test-backup`                                             |
| OCI     | OCID               | `ocid1.bucket.oc1.eu-frankfurt-1.aaaaaaaa5b5d7phlob22x4xin2lopq33ugriqiglek2ecxecrjx2awceb7eq`                                               |

How to form a CCRID:

>  * [AWS (EC2 Instance)][13]: `arn:aws:ec2:{region}:{accountId}:instance/{instanceId}`, To retrieve the `instanceId`:

Reviewer: ```suggestion
 * [AWS (EC2 Instance)][13]: `arn:aws:ec2:{region}:{accountId}:instance/{instanceId}`. 
    Use this command to retrieve the `instanceId`:
```

 ```bash
 ec2metadata --instance-id
 ```
 * [Azure][11]: `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}`
 * GCP: `//compute.googleapis.com/projects/{projectID}/zones/{zoneName}/instances/{instanceName}"`
 * OCI/Oracle: CCRID can be obtained by [sending a request][12] at: `http://169.254.169.254/opc/v2/instance/id`

For example, to set AWS CCRID for all resources in metrics, traces, and logs, use the [transform processor][2] with the following configuration:
```yaml
processors:
  transform:
    metric_statements:
      - context: resource
        statements:
          - set(attributes["datadog.ccrid"], "arn:aws:sns:us-east-1:123456789012:instance/example-sns-topic-name")
    trace_statements:
      - context: resource
        statements:
          - set(attributes["datadog.ccrid"], "arn:aws:sns:us-east-1:123456789012:instance/example-sns-topic-name")
    log_statements:
      - context: resource
        statements:
          - set(attributes["datadog.ccrid"], "arn:aws:sns:us-east-1:123456789012:instance/example-sns-topic-name")
```

> The OpenTelemetry semantic conventions also define the [cloud.resource_id][14] attribute , which can be mapped in the configuration using the [attributes processor][15].

Author: Potentially yes, but for the time being cloud.resource_id is supported only by Azure Logs Translator, Google Cloud Log Entry Encoding Extension, AWS WAF/ELB Log Unmarshaler components ... and level of this support is still quite drafty.
I would prefer to inform user explicitly that CCRID is in their hands, until we will be capable to make 100% transparent.
